examples/ecs/dynamic.rs

```diff
@@ -150,12 +150,12 @@ fn main() {
                 let mut query = builder.build();
 
                 query.iter_mut(&mut world).for_each(|filtered_entity| {
-                    #[allow(deprecated)]
                     let terms = filtered_entity
                         .access()
-                        .component_reads_and_writes()
-                        .0
-                        .map(|id| {
+                        .try_iter_component_access()
+                        .unwrap()
```

Comment:
Previously this example assumed that the iteration was always going to be over non-inverted access, which would silently misbehave at runtime if the assumption broke. Instead, the `unwrap()` explicitly crashes the program if that invalid state is reached.